src/coreclr/jit/codegencommon.cpp

```diff
@@ -1834,7 +1834,11 @@ void CodeGen::genGenerateMachineCode()
         printf(" for ");
 
 #if defined(TARGET_X86)
-        if (compiler->canUseEvexEncoding())
+        if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX10v1))
+        {
+            printf("X86 with AVX10v1");
```

Comment:
It would be nice if we could have this print `AVX10v1/256` or `AVX10v1/512` as appropriate, so that its understood if the hardware has 256-bit or 512-bit support.

I expect what we'd want to do is something more like:
```csharp
if (compiler->canUseEvexEncoding())
{
    if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX10v1))
    {
        if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F))
        {
            printf("X86 with AVX10/512");
        }
        else
        {
            printf("X86 with AVX10/256");
        }
    }
    else
    {
        assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F));
        printf("X86 with AVX512");
    }
}
```

I expect we don't need to actually print the `v1` or future versions, as the main consideration has just been the overall "baseline" we're targeting (i.e. why we just say `AVX512` and not `AVX512F+BW+CD+DQ+VL+...`).

Reply:
Shouldn't this be the following? They are both technically the same I think but the following looks better

if (compiler->canUseEvexEncoding())
{
    if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX10v1))
    {
        if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX10v1_V512))
        {
            printf("X86 with AVX10/512");
        }
        else
        {
            printf("X86 with AVX10/256");
        }
    }
    else
    {
        assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F));
        printf("X86 with AVX512");
    }
}

Reply:
I have made the suggested changes based on the discussion on teams.